docs/InDepth.md

```diff
@@ -0,0 +1,265 @@
+## Concept
+
+### Tagged Template Literals
+
+This library leverages [Tagged Template Literals](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals#Tagged_template_literals) interpolation functions.
+
+Template Literals allow **substitutions**.
+
+Normally, substitutions are passed **by value**. And such a function would look like this.
+
+**(BAD)**
+```javascript
+const tag = (template, ...substitutions) => ...
+const substitution = 10;
+
+tag`Some statement that can use ${substitution}`
+// => Some statement that can use 10
+```
+
+In this form we are not able to infer a sensible **token name**.
+
+Presuming we use the substitution index we would be left with a translation like this.
+
+```
+msgid "Some statement that can use __0__"
+msgstr ...
+```
+
+Obviously `__0__` is **not** a good placeholder. We need something more descriptive.
+
+We need to be equally explicit about the **token name** and **token value**.
+
+**(GOOD)**
+
+```javascript
+tag`Some statement that can use ${{num: substitution}}`
+// => Some statement that can use 10
+```
+
+We provide an `object` with single key-value and achieve a translation like this.
+
+```
+msgid "Some statement that can use __num__"
+msgstr ...
+```
+
+While this may feel a little contrived we feel it is the most explicit solution.
+
+### with plain values
+
+By default we require that all substitutions be object key-value as shown above.
+
+Any substitution passed **by value** will be treated like a normal Template Literal and will **appear in your translation**.
+
+```
+const substitution = 'foo';
+tag`Some simple ${substitution}`
+```
+
+Results in:
+
+```
+msgid "Some simple foo"
+msgstr ...
+```
+
+### with React
+
+Consider a React `Paginator` component where `More` and `Less` anchors will allow more or fewer results to be shown.
+
+```html
+<span>Show <a>10 More</a> or <a>5 Less</a></span>
+```
+
+The anchors themselves would need to be separate HTML elements, nested in the overall statement. And the number of more or less items may vary. So it is reasonable to assume they would be nested React components.
+
+In a perfect world we could use a **Tagged Template Literal** to compose idiomatically.
+
+```
+<span>
+  {tag`Show
+    ${{more: <More {...{numMore, onMore}} />} or
+    ${{less: <Less {...{numLess, onLess}} />}`}
+</span>
+```
+
+Presumably we will translate the `<More/>` and `<Less/>` components separately. But we will want our translation to still show some context.
+
+```
+msgid "Show __more__ or __less__"
+msgstr ...
+```
+
+We can achieve this if our interpolator function returns `Array` any time we have one or more non-string **token value**. React will then treat this array as `{children}`.
+
+So long as all elements have a `key` then React is happy. So we make sure we do this for you under the hood.
+
+
+## API
+
+The API is heavily influenced by [gettext](https://en.wikipedia.org/wiki/Gettext) translation package.
```

Comment:
Lets put this whole section at the top. It is good context.

Reply:
At the top of the main docs? I think it might be a bit repetitive because they begin with this atm:

`Intended for [gettext](https://www.npmjs.com/search?q=gettext) or similar translation APIs (where the key is the intended text in the developer's locale).`

Reply:
Good point. Well made.